### PR TITLE
chore(release): fix release script

### PR DESCRIPTION
**Pull request by xxchan**
Signed-off-by: xxchan <37948597+xxchan@users.noreply.github.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Tested with v0.1.12 release.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: ci/scripts/release.sh [hunk 1]
```diff
@@ -15,8 +15,8 @@ gh release create "${BUILDKITE_TAG}" --generate-notes -d -p
 
 echo "--- Download artifacts"
 mkdir -p target/debug && cd target/debug
-buildkite-agent artifact download risingwave-release .
-mv risingwave-release risingwave
+buildkite-agent artifact download risingwave-ci-release .
```
Comment (BugenZhao) [inline ci/scripts/release.sh:18]: `ci-release` will include debug assertions and debug info. Is this expected for releases, to external users?

Comment (skyzh) [reply ci/scripts/release.sh:18]: I remembered that ci-release is building RisingWave in debug mode while others in release mode?

Comment (BugenZhao) [reply ci/scripts/release.sh:18]: > building RisingWave in debug mode while others in release mode

That's ci-dev. 😍

https://github.com/risingwavelabs/risingwave/blob/3e0d4d2873c47c1bd141ccd90d286c75fff06e6f/Cargo.toml#L61-L78

Comment (skyzh) [reply ci/scripts/release.sh:18]: We want debug info in our binary. So that if it crashes somewhere, it would be easier for us to read the stack trace.

Comment (BugenZhao) [reply ci/scripts/release.sh:18]: Okay. So I believe overflow checks are also necessary (an example: #5196). Not sure about debug assertions. 🤣

Comment (skyzh) [reply ci/scripts/release.sh:18]: Oh, we don't need debug assertions. So we need a new profile for release builds right? Or we just need to use the original "release" profile from cargo?

Comment (BugenZhao) [reply ci/scripts/release.sh:18]: Yes. The `release` profile is untouched.

**Comment by huangjw806** [comment]
To use earlier glibc versions, we now build risingwave directly inside the Release script. #5414

**Comment by skyzh** [comment]
Thanks!

**Pull request closed without merging**